pyv/transpiler.py

```diff
@@ -0,0 +1,593 @@
+            )
+        else:
+            return super().visit_BinOp(node)
+
+    def visit_ClassDef(self, node):
+        extractor = DeclarationExtractor(VTranspiler())
+        extractor.visit(node)
+        declarations = node.declarations = extractor.get_declarations()
+        node.class_assignments = extractor.class_assignments
+        ret = super().visit_ClassDef(node)
+        if ret is not None:
+            return ret
+
+        fields = []
+        index = 0
+        for declaration, typename in declarations.items():
+            if typename == None:
+                typename = "ST{0}".format(index)
+                index += 1
+            fields.append(f"{declaration}: {typename}")
+
+        for b in node.body:
+            if isinstance(b, ast.FunctionDef):
+                b.self_type = node.name
+
+        object_def = "type\n"
+        object_def += self.indent(f"{node.name} = object\n", level=node.level + 1)
+        object_def += "\n".join([self.indent(f, level=node.level + 2) for f in fields])
+        body = [self.visit(b) for b in node.body]
+        body = "\n".join(body)
+        return f"{object_def}\n{body}\n"
+
+    def visit_IntEnum(self, node):
+        fields = []
+        for member, var in node.class_assignments.items():
+            var = self.visit(var)
+            if var == "auto()":
+                fields.append(f"{member},")
+            else:
+                fields.append(f"{member} = {var},")
+        fields = "\n".join([self.indent(f) for f in fields])
+        return f"type {node.name} = enum\n{fields}\n\n"
+
+    def visit_IntFlag(self, node):
+        fields = []
+        for member, var in node.class_assignments.items():
+            var = self.visit(var)
+            if var == "auto()":
+                fields.append(f"{member},")
+            else:
+                fields.append(f"{member} = {var},")
+        fields = "\n".join([self.indent(f, level=2) for f in fields])
+        # flags = self.indent(f"{node.name}Flags = set[{node.name}]")
+        return "\n".join(
+            [
+                "type",
+                self.indent(f"{node.name} = enum"),
+                f"{fields}",
+                # f"{flags}",
+                "",
+            ]
+        )
+
+    def visit_StrEnum(self, node):
+        fields = []
+        for member, var in node.class_assignments.items():
+            var = self.visit(var)
+            if var == "auto()":
+                fields.append(f"{member},")
+            else:
+                fields.append(f"{member} = {var},")
+        fields = "\n".join([self.indent(f) for f in fields])
+        return f"type {node.name} = enum\n{fields}\n\n"
+
+    def visit_List(self, node):
+        elements = [self.visit(e) for e in node.elts]
+        elements = ", ".join(elements)
+        return f"[{elements}]"
+
+    def visit_Set(self, node):
+        raise AstNotImplementedError("Sets are not implemented in V yet.")
```

Comment:
While generating `global2.v`: `global2.py:10:6: py2many.exceptions.AstNotImplementedError: __init__() missing 1 required positional argument: 'node'`.

Pass `node` as second arg to `AstNotImplementedError`.